src/Form/Form.tsx

```diff
@@ -325,6 +326,7 @@ const Form: FormComponent = React.forwardRef((props: FormProps, ref: React.Ref<F
        * so use Ref to get future value
        */
       const formValue = omit(realFormValueRef.current, [name]);
+      realFormValueRef.current = formValue;
```

Comment:
The reason isn't this sentence. it's necessary. 
And look at this [example](https://codesandbox.io/s/keen-jackson-7ikpkw?file=/src/index.js).
when "email" is true. "username" and "password" will unmount, and then React will call "removeField", But because the calls are continuous, so the latter will overwrite the earlier one.